packages/editor/src/components/browser/TypeBrowser.tsx

```diff
@@ -13,40 +14,88 @@ export const useTypeBrowser = (onDoubleClick: () => void): UseBrowserImplReturnV
     id: TYPE_BROWSER_ID,
     name: 'Type',
     content: <TypeBrowser value={value} onChange={setValue} onDoubleClick={onDoubleClick} />,
-    accept: () => value
+    accept: () => value,
+    icon: IvyIcons.DataClass
   };
 };
 
 const TypeBrowser = (props: { value: string; onChange: (value: string) => void; onDoubleClick: () => void }) => {
+  type TypeBrowserObject = JavaType & {
+    icon: IvyIcons;
+  };
+
   const { context } = useEditorContext();
 
+  const [allSearchActive, setAllSearchActive] = useState(false);
+
   const [mainFilter, setMainFilter] = useState('');
   const { data: allDatatypes, isFetching } = useMeta('meta/scripting/allTypes', { context, limit: 150, type: mainFilter }, []);
+  const dataClasses = useMeta('meta/scripting/dataClasses', context, []).data;
+  const ivyTypes = useMeta('meta/scripting/ivyTypes', undefined, []).data;
 
-  const [dynamicTypes, setDynamicTypes] = useState<JavaType[]>([]);
+  const [types, setTypes] = useState<TypeBrowserObject[]>([]);
 
   const [typeAsList, setTypeAsList] = useState(false);
 
   const [showHelper, setShowHelper] = useState(false);
 
   useEffect(() => {
-    if (mainFilter.length > 0) {
-      allDatatypes.sort((a, b) => a.simpleName.localeCompare(b.simpleName));
+    const sortAndMapTypes = (types: TypeBrowserObject[]): TypeBrowserObject[] => {
+      return types.slice().sort((a, b) => {
+        const fqCompare = a.fullQualifiedName.localeCompare(b.fullQualifiedName);
+        if (fqCompare !== 0) {
+          return fqCompare;
+        }
+        return a.simpleName.localeCompare(b.simpleName);

```

Comment:
of course you can also create on function that do the sorting for you. I expected only the sorting function itself as shared function, but ok

```ts
const typeComparator = (a: TypeBrowserObject, b: TypeBrowserObject) => {
  const fqCompare = a.fullQualifiedName.localeCompare(b.fullQualifiedName);
  if (fqCompare !== 0) {
    return fqCompare;
  }
  return a.simpleName.localeCompare(b.simpleName);
}

const sortedIvyTypes = mappedIvyTypes.sort(typeComparator);
const sortedMappedDataClasses = mappedDataClasses.sort(typeComparator);
```